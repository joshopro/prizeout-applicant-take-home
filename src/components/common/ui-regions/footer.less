@import '../../../utils/styles/mixins.less';
@import '../../../utils/styles/media-queries.less';

.footer {
	padding: var(--space-s) var(--space-inner);
	display: grid;
	gap: var(--space-m);
	align-items: center;
	grid-template-columns: auto 1fr;
	grid-template-areas:
		"copyright links";
	
	@media @desktop-down {
		grid-template-columns: 1fr;
		grid-template-areas:
			"copyright"
			"links";
		justify-items: center;
		text-align: center;
	}
	
	&__copyright {
		justify-self: start;
	}
	&__links {
		justify-self: end;
		display: flex;
		flex-flow: row wrap;
		align-items: center;
		justify-content: center;
		padding: 0;
		margin: 0;
		list-style-type: none;
		
		@media @desktop-down {
			justify-self: center;	
		}
	}

	&__link {
		margin: 0;
		list-style-type: none;
		
		a {
			margin: 0 var(--space-xs);
			@media @mobile-portrait-down {
				margin: 0 .05rem;
			}
		}
	}
}
