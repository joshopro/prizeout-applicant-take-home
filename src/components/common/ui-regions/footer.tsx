import React from 'react';

import './footer.less';

const website_domain = process.env['ALTERNATE_DOMAIN'];
const current_year = new Date().getFullYear();

export const Footer: React.FC = (): React.ReactElement => {
    return (
        <footer className="footer">
            <div className="footer_copyright">
                <p>&copy; 2020-{current_year}, Prizeout, Inc. All rights reserved.</p>
            </div>
            <ul className="footer__links">
                <li className="footer__link">
                    <a href={`https://${website_domain}/privacy/`} target="_blank" rel="noreferrer">
                        Privacy Policy
                    </a>
                    &nbsp;|&nbsp;
                </li>
                <li className="footer__link">
                    <a href={`https://${website_domain}/terms/`} target="_blank" rel="noreferrer">
                        Prizeout Terms of Use
                    </a>
                    &nbsp;|&nbsp;
                </li>
                <li className="footer__link">
                    <a
                        href={`https://${website_domain}/privacy/index.html#infocollect`}
                        target="_blank"
                        rel="noreferrer"
                    >
                        California Info We Collect
                    </a>
                    &nbsp;|&nbsp;
                </li>
                <li className="footer__link">
                    <a
                        href="https://privacyportal.cookiepro.com/webform/20dcda57-44d2-4622-b07c-30902b02fbd4/9211ec70-941f-4529-9e43-027527ed5e2f"
                        target="_blank"
                        rel="noreferrer"
                    >
                        Do Not Sell My Personal Information
                    </a>
                </li>
            </ul>
        </footer>
    );
};
