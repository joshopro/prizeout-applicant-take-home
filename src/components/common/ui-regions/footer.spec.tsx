import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';

import { Footer } from './footer';

describe('Test Footer Component', () => {
    test('Component matches snapshot', () => {
        const { asFragment } = render(<Footer />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('No TOS is shown unless on gift card details page', () => {
        render(<Footer />);

        expect(screen.queryByText('Terms and Conditions')).not.toBeInTheDocument();
    });
});
