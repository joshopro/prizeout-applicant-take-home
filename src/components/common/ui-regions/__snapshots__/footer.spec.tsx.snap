// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer Component Component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footer_copyright"
    >
      <p>
        © 2020-2022, Prizeout, Inc. All rights reserved.
      </p>
    </div>
    <ul
      class="footer__links"
    >
      <li
        class="footer__link"
      >
        <a
          href="https://undefined/privacy/"
          rel="noreferrer"
          target="_blank"
        >
          Privacy Policy
        </a>
         | 
      </li>
      <li
        class="footer__link"
      >
        <a
          href="https://undefined/terms/"
          rel="noreferrer"
          target="_blank"
        >
          Prizeout Terms of Use
        </a>
         | 
      </li>
      <li
        class="footer__link"
      >
        <a
          href="https://undefined/privacy/index.html#infocollect"
          rel="noreferrer"
          target="_blank"
        >
          California Info We Collect
        </a>
         | 
      </li>
      <li
        class="footer__link"
      >
        <a
          href="https://privacyportal.cookiepro.com/webform/20dcda57-44d2-4622-b07c-30902b02fbd4/9211ec70-941f-4529-9e43-027527ed5e2f"
          rel="noreferrer"
          target="_blank"
        >
          Do Not Sell My Personal Information
        </a>
      </li>
    </ul>
  </footer>
</DocumentFragment>
`;
